File: backend/src/main/java/com/condos/incidente/dto/CreateIncidenteRequest.java
package com.condos.incidente.dto;

import com.condos.incidente.model.PrioridadIncidente;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class CreateIncidenteRequest {

    @NotBlank
    private String categoria;

    @NotBlank
    private String titulo;

    @NotBlank
    private String descripcion;

    @NotBlank
    private String ubicacion;

    @NotNull
    private PrioridadIncidente prioridad;
}

File: lib/features/incidentes/models/create_incidente_request.dart
import 'package:json_annotation/json_annotation.dart';
import 'incidente.dart';

part 'create_incidente_request.g.dart';

@JsonSerializable()
class CreateIncidenteRequest {
  final String categoria;       // nombre dinámico de la categoría
  final String titulo;
  final String descripcion;
  final String ubicacion;
  final PrioridadIncidente prioridad;

  CreateIncidenteRequest({
    required this.categoria,
    required this.titulo,
    required this.descripcion,
    required this.ubicacion,
    required this.prioridad,
  });

  factory CreateIncidenteRequest.fromJson(Map<String, dynamic> json) =>
      _$CreateIncidenteRequestFromJson(json);
  Map<String, dynamic> toJson() => _$CreateIncidenteRequestToJson(this);
}
